llvm/lib/Target/AIE/AIELegalizerHelper.cpp

```diff
@@ -53,8 +53,9 @@ bool AIELegalizerHelper::pack32BitVector(LegalizerHelper &Helper,
   const LLT RegTy = MRI.getType(DstReg);
   while (Operand != OperandEnd) {
     Register DestinationOperand = Operand->getReg();
+    const LLT DstOpTy = MRI.getType(DestinationOperand);
 
-    if (RegTy.getScalarSizeInBits() != 32) {
+    if (RegTy.getScalarSizeInBits() != 32 && DstOpTy.getSizeInBits() != 32) {
```

Comment:
I still don't get why we have to verify this now? What changed in the code?

Reply:
Earlier this `pack32BitVector` used by G_BUILD_VECTOR where inputs are `s8` or `s16`, here they were zero extended. But for G_BUILD_VECTOR_TRUNC where the inputs are `s32`, there is no necessity for ZEXT

Reply:
Could we remove `RegTy.getScalarSizeInBits() != 32` then? This seems redundant, as we check a "safer" predicate now.